src/darwin/Framework/CHIP/MTREndpointInfo.mm

```diff
@@ -0,0 +1,210 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "MTREndpointInfo_Internal.h"
+
+#import "MTRAttributeTLVValueDecoder_Internal.h"
+#import "MTRDeviceTypeRevision.h"
+#import "MTRLogging_Internal.h"
+#import "MTRStructsObjc.h"
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+
+#include <deque>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+
+enum class EndpointMark : uint8_t {
```

Comment:
I wonder whether this should be EndpointTraversalState, and the field be _traversalState?

Reply:
it's used slightly differently in the two passes, so I think the generic "mark" is fine.